abnumber/alignment.py

```diff
@@ -1,4 +1,4 @@
-from typing import Union
```

Comment:
Overall, the code is well-written and clear. However, there are a few recommendations for improvement:

1. **Type Hints**: It would be helpful to include type hints for all function parameters and return types to improve readability and maintainability. For example, in the `format` function, specify the type for `mark_identity` and `mark_cdrs`:

```python
def format(self, mark_identity: bool = True, mark_cdrs: bool = True) -> str:
```

2. **Docstrings**: Some functions could use more detailed docstrings. For instance, the `slice` function could benefit from explaining what the `allow_raw` parameter does.

```python
def slice(self, start: Union[str, int, 'Position', None] = None, stop: Union[str, int, 'Position', None] = None,
              stop_inclusive: bool = True, allow_raw: bool = False):
    """Create a slice of this alignment

    :param start: Slice start position (inclusive), :class:`Position` or string (e.g. '111A')
    :param stop: Slice stop position (inclusive), :class:`Position` or string (e.g. '112A')
    :param stop_inclusive: Include stop position in slice
    :param allow_raw: Allow unaligned numeric indexing from 0 to length of sequence - 1. If set to False, only named position indexing is allowed.
    :return: new sliced Alignment object
    """
```

3. **Error Handling**: In the `__getitem__` method, it would be more informative to raise a `ValueError` instead of an `IndexError` when the step of the slice is not 1. This is because the issue is with the value of the step, not with the index itself.

```python
if item.step is not None and item.step != 1:
    raise ValueError(f'Slicing with step != 1 is not implemented, got: {item}')
```

4. **Code Duplication**: There's some repeated code in the `__getitem__` and `slice` methods where the start and stop positions are parsed. Consider refactoring this into a separate method.